analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp11.cs

```diff
@@ -8,3 +8,32 @@ public interface IInterface
     [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
     public static extern virtual bool RemoveDirectory2(string name);  // Noncompliant
 }
+
+namespace LibraryImportAttributeTests
+{
+    // Copy the class to a .Net 7 project in VS and use "goto definition" on the methods to extract the code generated by the Microsoft.Interop.LibraryImportGenerator
+    public static partial class ExternMethods
+    {
+        [LibraryImport("foo.dll")]
+        public static partial void DllImportAttributeAppliedToThisFunction(); // Noncompliant
```

Comment:
nitpick: should these be renamed `LibraryImportAttributeApplied...` instead of `DllImport`? because it's slightly confusing

Reply:
No. The DllImport attribute gets applied by the LibraryImport source generator to this method. The actual method after the source generator run is the combination of the two partial definitions:
```cs
        [LibraryImport("foo.dll")]
        [System.Runtime.InteropServices.DllImportAttribute("foo.dll", EntryPoint = "DllImportAttributeAppliedToThisFunction", ExactSpelling = true)]
        public static extern partial void DllImportAttributeAppliedToThisFunction(); 
```